### PR TITLE
Feature/add send sync

### LINKED ISSUE #74: Consider making the error Sync + Send
Currently the [error](https://docs.rs/wascc-host/0.11.0/wascc_host/errors/struct.Error.html) type used is neither `Send` nor `Sync`. I believe that top-level error types like these should always be Send + Sync. This allows for greater flexibility on the handling of errors as well as libraries that could be used (for example the `anyhow` crate needs its errors to be `Send + Sync`.

Is there a specific reason why they cannot be `Send + Sync`? I have looked through the code and parts of it seems to stem from the `wapc` crate. However I did not find a good reason for its existence, as it [is not used](https://github.com/wapc/wapc-rust/search?q=HostCallFailure&unscoped_q=HostCallFailure) there.

What do you think? I'd gladly cook up a PR to make this change.

### PR DESCRIPTION
**Pull request by TheNeikos**
This PR makes the public error type `Send + Sync` and fixes #74

### CODE REVIEW

**Comment by TheNeikos** [comment]
Should I also bump the version for this one?

File: Cargo.toml [hunk 1]
```diff
@@ -27,8 +27,8 @@ crossbeam-utils = "^0.7.0"
 prometheus = { version = "0.9", features = ["push"], optional = true }
 hyper = { version = "0.13", optional = true }
 tokio = { version = "0.2", features = ["macros"], optional = true }
-wapc = "0.8.1"
-wascc-codec = "0.7.2"
+wapc = "0.9"
+wascc-codec = "0.8"
```
Comment (TheNeikos) [inline Cargo.toml:31]: Note: This won't build until wascc-codec has been released

**Review by autodidaddict — CHANGES_REQUESTED** [review]
Bump lattice-client dep to 0.0.3  -  lattice client also depends on wascc-codec, so we had conflicting transitive deps.

File: Cargo.toml (hunk 1 above)
Comment (autodidaddict) [reply Cargo.toml:31]: There's a transitive dependency problem on wascc-codec that won't be fixed until the lattice-client dependency gets bumped to 0.0.3

**Pull request merged**